### ISSUE TITLE
Permission Revamp [Feedback]

### ISSUE THREAD

**Issue by JohnnyKing94** [translate/pootle#4962]
About the permissions page, i have some feedbacks:

- Custom groups (in this way the admins do not to give the SAME powers to specific translators). 
- New permisionss, like: 
-- Who can comment (included in this "who can edit other comments", "edit my own comment", "edit time restriction", "who can delete other comments", "delete own comment"
-- Who can report (included in this "Report flood control")
-- Who can silence the checks
-- Who can download the project file (i don't like that all default users can download the project)
-- Who can upload the project file
-- Search flood control (i want to set a time for specific groups to search specific translations)

- Remove "Can perform administrative tasks" and give more specific powers for administration control. For instance, i want to give the admin powers to someone but i do not like to give all full controll access of the CP. Therefore try to "modularize" all the tasks that an admin can do (so all the tabs like Info, Permissions, Users, Reports, Pages etc need to be individually assigned)

If you take in consideration these ideas, Pootle will be the best localization platform ever made... If i have other ideas about revamping the permision, i'll add below

**Comment by unho** [comment]
Yes, indeed permissions can benefit from some changes. I always missed the ability to see all the permissions for a single user, or the possibility to have roles.

Regarding "Can perform administrative tasks" I think it only applies to outside the control panel (CP or admin dashboard) since there only superusers can log. In other words, it only gives a restricted administration access to the objects the user has this permission, such as some specific language, project or translation project.

**Comment by unho** [comment]
I just tried to check who can review suggestions for Urdu language in MLO server, and given that a lot of users have permissions assigned for that language it was a bit difficult to easily spot.

I think permissions page can be tweaked a bit to list which users have each permission, like Administrators (Language Coordinators) section and then list the users with administrate permission, then Reviewers section and list the reviewers and so on. In some way the permissions that can be assigned for a specific language are kinda incremental: suggest, translate, review, administrate. So in some way they can almost be considered roles.

Right now the Permissions tab can only be seen by users with a given level of permissions, since what you can do there is alter the permissions. But in MLO we are redirecting new translators all the time because they are unable to contact their language coordinators directly. So it would be great to expose the Permissions list (a static one listing users by their roles) so new translators can easily find who is who for their language.
